src/simple/room/Participant.js

```diff
@@ -46,28 +51,68 @@ class Participant extends Emitter {
     this.screensharing = false;
     this.extra = extra;
 
-    this.ON_UPDATED = 'ON_UPDATED';
-    this.ON_START_TALKING = 'ON_PARTICIPANT_START_TALKING';
-    this.ON_STOP_TALKING = 'ON_PARTICIPANT_STOP_TALKING';
-    this.ON_DISCONNECT = 'ON_PARTICIPANT_DISCONNECT';
-    this.ON_STREAM_SUBSCRIBED = 'ON_STREAM_SUBSCRIBED';
-    this.ON_STREAM_UNSUBSCRIBED = 'ON_STREAM_UNSUBSCRIBED';
-    this.ON_AUDIO_MUTED = 'ON_AUDIO_MUTED';
-    this.ON_AUDIO_UNMUTED = 'ON_AUDIO_UNMUTED';
-    this.ON_VIDEO_MUTED = 'ON_VIDEO_MUTED';
-    this.ON_VIDEO_UNMUTED = 'ON_VIDEO_UNMUTED';
-    this.ON_SCREENSHARING = 'ON_SCREENSHARING';
-    this.ON_STOP_SCREENSHARING = 'ON_STOP_SCREENSHARING';
+    this.ON_UPDATED = 'participant/ON_UPDATED';
+    this.ON_START_TALKING = 'participant/ON_START_TALKING';
+    this.ON_STOP_TALKING = 'participant/ON_STOP_TALKING';
+    this.ON_DISCONNECT = 'participant/ON_DISCONNECT';
+    this.ON_STREAM_SUBSCRIBED = 'participant/ON_STREAM_SUBSCRIBED';
+    this.ON_STREAM_UNSUBSCRIBED = 'participant/ON_STREAM_UNSUBSCRIBED';
+    this.ON_AUDIO_MUTED = 'participant/ON_AUDIO_MUTED';
+    this.ON_AUDIO_UNMUTED = 'participant/ON_AUDIO_UNMUTED';
+    this.ON_VIDEO_MUTED = 'participant/ON_VIDEO_MUTED';
+    this.ON_VIDEO_UNMUTED = 'participant/ON_VIDEO_UNMUTED';
+    this.ON_SCREENSHARING = 'participant/ON_SCREENSHARING';
+    this.ON_STOP_SCREENSHARING = 'participant/ON_STOP_SCREENSHARING';
+    this.ON_EXTRA_CHANGE = 'participant/ON_EXTRA_CHANGE';
   }
 
   triggerEvent(name: string, ...args: any[]) {
     this.eventEmitter.emit.apply(this.eventEmitter, [name, ...args]);
-    this.eventEmitter.emit(this.ON_UPDATED);
+    this.eventEmitter.emit.apply(this.eventEmitter, [this.ON_UPDATED, ...args]);
   }
 
-  onTalking(isTalking: boolean) {
+  triggerUpdate(eventType: string, broadcast: boolean = true) {
+    const status: Object = { callId: this.callId };
+
+    switch (eventType) {
+      case this.ON_START_TALKING:
+      case this.ON_STOP_TALKING: {
+        status.isTalking = this.isTalking;
+        break;
+      }
+      case this.ON_AUDIO_MUTED:
+      case this.ON_AUDIO_UNMUTED: {
+        status.audioMuted = this.audioMuted;
+        break;
+      }
+      case this.ON_VIDEO_MUTED:
+      case this.ON_VIDEO_UNMUTED: {
+        status.videoMuted = this.videoMuted;
+        break;
+      }
+      case this.ON_SCREENSHARING:
+      case this.ON_STOP_SCREENSHARING: {
+        status.screensharing = this.screensharing;
+        break;
+      }
+      case this.ON_EXTRA_CHANGE: {
+        status.extra = this.extra;
+        break;
+      }
+      default:
+    }
+
+    if (broadcast) {
+      this.broadcastStatus(status);
+    }
+
+    this.eventEmitter.emit.apply(this.eventEmitter, [eventType, status]);
```

Comment:
`this.eventEmitter.emit(eventType, status);` no need to `appy` here as you know all the args.